cpp/include/cudf/detail/aggregation/aggregation.hpp

```diff
@@ -27,6 +27,63 @@
 
 namespace cudf {
 namespace detail {
+
+class aggregation_finalizer;
+
+class compound_aggregation {
+ public:
+  virtual std::vector<aggregation> get_simple_aggregations(data_type col_type) const { return {}; }
```

Comment:
Should this be pure virtual? When is an empty `vector<aggregation>` acceptable?

Reply:
fixed.